### PR TITLE
wip: bring WT integration tests up to spec

### PR DESCRIPTION
**Pull request by robercano**
## Description
This PR focuses on significantly increasing test coverage across core protocol components, particularly for the Whitelisting logic, Fleet Commanders, and the WisdomTree Ark integration. It introduces extensive test suites for various edge cases, unhappy paths, and authorizations. Additionally, it contains some routine cleanup of obsolete contracts in the codebase.

## Changes
- **New Test Files Added:**
  - `AdmiralsQuartersWhitelist.t.sol` and `AdmiralsQuartersWhitelist.permit.t.sol` (>800 lines of comprehensive tests for `AdmiralsQuartersWhitelist`).
- **Test Coverage Expanded For:**
  - `WisdomTreeArk.t.sol` (+144 lines)
  - `FleetCommanderWhitelist.t.sol` (+304 lines)
  - `ERC4626MultiToken.t.sol` 
  - `RoundsVaultTwoPhaseSettlement.t.sol`
  - `Whitelist.t.sol` 
- **Contract Updates:**
  - Refactored/improved missing logic and interfaces for `AdmiralsQuartersWhitelist` based on discoveries during testing test-driven development.
  - Deleted legacy `RoundsVaultExchanger.sol` which is no longer needed.

## Benefits
1. **Improved Reliability:** Greatly increases protocol confidence by ensuring high test coverage across `FleetCommanderWhitelist`, `AdmiralsQuartersWhitelist`, and `WisdomTreeArk`.
2. **Robust Access Control:** Deeply tests whitelisting functionality and permit configurations, eliminating potential access loopholes.
3. **Reduced Tech Debt:** Cleans up unused code like `RoundsVaultExchanger.sol` to maintain a lean repository.

## Testing
- Extended the Foundry test suite with exhaustive unit tests targeting edge cases, failure states, and exact revert outcomes.
- Run `forge test` to execute the full suite.
- Run `forge coverage` to verify the coverage increases across the newly targeted files.

Please review and provide any feedback or suggestions for improvement.

### CODE REVIEW

File: packages/access-contracts/test/ProtocolAccessManagedV2.t.sol [hunk 1]
```diff
@@ -13,6 +14,14 @@ contract MockFleetV2 is ProtocolAccessManagedV2 {
     function restrictedAction() external onlyOperator returns (bool) {
         return true;
     }
+
+    function getAccessManagerV2()
```
Comment (halaprix) [inline packages/access-contracts/test/ProtocolAccessManagedV2.t.sol]: this can be removed. looks like `_getAccessManagerV2()` is not used anywhere

File: packages/access-contracts/test/ProtocolAccessManagedV2.t.sol [hunk 2]
```diff
@@ -75,4 +84,8 @@ contract ProtocolAccessManagedV2Test is Test {
         assertTrue(fleet.hasOperatorRole(operator));
         assertFalse(fleet.hasOperatorRole(user));
     }
+
+    function test_GetAccessManagerV2() public view {
```
Comment (halaprix) [inline packages/access-contracts/test/ProtocolAccessManagedV2.t.sol]: this can be removed. looks like `_getAccessManagerV2()` is not used anywhere

File: packages/core-contracts/src/contracts/AdmiralsQuartersWhitelist.sol [hunk 3]
```diff
@@ -158,6 +162,40 @@ contract AdmiralsQuartersWhitelist is
         emit FleetEntered(_msgSender(), fleetCommander, assets, shares);
     }
 
+    /// @inheritdoc IAdmiralsQuartersWhitelist
+    function enterFleetWithPermit2(
```
Comment (halaprix) [inline packages/core-contracts/src/contracts/AdmiralsQuartersWhitelist.sol:209]: why only `enterFleetWithPermit2` and not `enterFleetWithPermit` ?

File: packages/core-contracts/test/admiralsQuarters/AdmiralsQuartersWhitelist.t.sol [hunk 4]
```diff
@@ -0,0 +1,786 @@
...
+        admiralsQuarters = new AdmiralsQuartersWhitelist(
+            ONE_INCH_ROUTER,
+            address(configurationManager),
+            address(accessManager),
+            WETH
+        );
+
+        vm.startPrank(governor);
+        // Grant Operator Role to AQ on the Fleet
+        accessManager.grantOperatorRole(
```
Comment (halaprix) [inline packages/core-contracts/test/admiralsQuarters/AdmiralsQuartersWhitelist.t.sol:97]: I'd add a test with operator revoked and whitelisted users.

File: packages/core-contracts/test/fleets/FleetCommanderWhitelist.t.sol [hunk 5]
```diff
@@ -378,4 +381,316 @@ contract FleetCommanderWhitelistTest is
...
+        whitelistFleet.removeArk(ark);
+        vm.stopPrank();
+
+        assertFalse(whitelistFleet.isArkActiveOrBufferArk(ark));
+        assertEq(whitelistFleet.getActiveArks().length, 0);
+    }
+
+    function test_DirectWrapperCalls_Coverage() public {
+        vm.startPrank(operator);
+        // We deposited "amount" for operator in setUp. Wait, operator has amount*2 minted and deposited "amount" for operator and "amount" for mockUser.
```
Comment (halaprix) [inline packages/core-contracts/test/fleets/FleetCommanderWhitelist.t.sol]: remove the `llm comments` -> they know what to do ;)

File: packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol [hunk 6]
```diff
@@ -428,4 +428,69 @@ contract RoundsVaultTwoPhaseSettlementTest is
...
+        vm.prank(keeper);
+        inputVault.nextRound(); // Round 0 is now InSettlement, Round 1 is Opened
+
+        assertEq(
+            uint256(inputVault.roundState(0)),
+            uint256(RoundState.InSettlement)
+        );
+
+        // Use a prank with proper admin check.
+        // Governor in testing is usually the deployer or an explicit address. By default MockAccessManager needs ADMIN_ROLE or something, let's see.
```
Comment (halaprix) [inline packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol]: same here

File: packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol [hunk 7]
```diff
@@ -428,4 +428,69 @@ contract RoundsVaultTwoPhaseSettlementTest is
...
+        // ProtocolAccessManagedV2 `onlyGovernor` modifier checks `IProtocolAccessManagerV2.hasRole(GOVERNOR_ROLE, msg.sender)`.
+        bytes32 GOVERNOR_ROLE = accessManager.GOVERNOR_ROLE();
+        address gov = address(0x909);
```
Comment (halaprix) [inline packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol]: this could be moved to `setUp()` - governor is reused across tests

File: packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol [hunk 8]
```diff
@@ -428,4 +428,69 @@ contract RoundsVaultTwoPhaseSettlementTest is
...
+
+    function test_EmergencyRollbackRound_RevertsNotGovernor() public {
+        vm.prank(userA);
+        inputVault.deposit(100 ether, userA);
+
+        vm.prank(keeper);
+        inputVault.nextRound();
+
+        vm.startPrank(userA);
+        vm.expectRevert(); // Typically AccessControlUnauthorizedAccount or something similar in ProtocolAccessManagedV2
```
Comment (halaprix) [inline packages/core-contracts/test/rounds-vault/RoundsVaultTwoPhaseSettlement.t.sol]: this should use the error selector

**Review by halaprix — CHANGES_REQUESTED** [review]
added some comments

File: packages/core-contracts/src/contracts/AdmiralsQuartersWhitelist.sol (hunk 3 above)
Comment (robercano) [reply packages/core-contracts/src/contracts/AdmiralsQuartersWhitelist.sol:209]: I thought we would only use the permit2. I'll add the first one as well

**Review by halaprix — APPROVED** [review]
looks good!
